src/storage/admin/RebuildIndexTask.cpp

```diff
@@ -18,15 +18,32 @@ RebuildIndexTask::genSubTasks() {
     space_ = ctx_.parameters_.space_id;
     auto parts = ctx_.parameters_.parts;
 
-    auto itemsRet = getIndexes(space_);
-    if (!itemsRet.ok()) {
-        LOG(ERROR) << "Indexes Not Found";
-        return cpp2::ErrorCode::E_INDEX_NOT_FOUND;
+    IndexItems items;
+    if (ctx_.parameters_.task_specfic_paras.empty()) {
+        auto itemsRet = getIndexes(space_);
+        if (!itemsRet.ok()) {
+            LOG(ERROR) << "Indexes not found";
+            return cpp2::ErrorCode::E_INDEX_NOT_FOUND;
+        }
+
+        items = std::move(itemsRet).value();
+    } else {
+        std::vector<std::string> indexes;
+        folly::split(",", ctx_.parameters_.task_specfic_paras[0], indexes, true);
+
+        for (const auto& index : indexes) {
+            auto indexID = folly::to<IndexID>(index);
+            auto indexRet = env_->indexMan_->getTagIndex(space_, indexID);
+            if (!indexRet.ok()) {
+                LOG(ERROR) << "Index not found";
+                return cpp2::ErrorCode::E_INDEX_NOT_FOUND;
+            }
+            items.emplace_back(indexRet.value());
+        }
     }
 
-    auto items = itemsRet.value();
     if (items.empty()) {
-        LOG(INFO) << "Index is empty";
+        LOG(ERROR) << "Index is empty";
```

Comment:
LOG(WARNING) ?

Reply:
the index id not found seems Error maybe better